google-cloud-testing/spanner-benchwrapper/src/main/java/com/google/cloud/benchwrapper/SpannerBenchWrapperImpl.java

```diff
@@ -0,0 +1,133 @@
+import com.google.cloud.spanner.ResultSet;
+import com.google.cloud.spanner.Spanner;
+import com.google.cloud.spanner.SpannerOptions;
+import com.google.cloud.spanner.Statement;
+import com.google.cloud.spanner.ReadOnlyTransaction;
+import com.google.cloud.spanner.Mutation;
+import com.google.cloud.spanner.SpannerBatchUpdateException;
+import com.google.cloud.spanner.TransactionContext;
+
+class SpannerBenchWrapperImpl extends SpannerBenchWrapperImplBase {
+  private Spanner spanner;
+  private DatabaseClient dbClient;
+
+  public SpannerBenchWrapperImpl(String spannerEmulatorHost) {
+    SpannerOptions.Builder builder = SpannerOptions.newBuilder()
+      .setHost("http://" + spannerEmulatorHost)
+      .setProjectId("test-project-id")
+      // Channels are secure by default (via SSL/TLS). For the example we disable TLS to avoid
+      // needing certificates.
+      .setChannelConfigurator(
+        new ApiFunction<ManagedChannelBuilder, ManagedChannelBuilder>() {
+          @Override
+          public ManagedChannelBuilder apply(ManagedChannelBuilder builder) {
+            return builder.usePlaintext();
+          }
+        }
+      );
+
+    SpannerOptions options = builder
+      .build();
+    spanner = options.getService();
+
+    dbClient = spanner.getDatabaseClient(
+      DatabaseId.of(options.getProjectId(), "test-instance", "test-db"));
+  }
+
+  public void read(ReadQuery request, StreamObserver<EmptyResponse> responseObserver) {
+    System.out.println("read has been called");
+
+    try (ReadOnlyTransaction transaction = dbClient.readOnlyTransaction()) {
+      ResultSet resultSet = transaction.executeQuery(Statement.of(request.getQuery()));
+      while (resultSet.next()) {
+        // Do nothing with the data.
+      }
+    }
+
+    EmptyResponse reply = EmptyResponse.newBuilder().build();
+    responseObserver.onNext(reply);
+    responseObserver.onCompleted();
+  }
+
+  public void insert(InsertQuery request, StreamObserver<EmptyResponse> responseObserver) {
+    System.out.println("insert has been called");
+
+    List<Mutation> mutations = new ArrayList<>();
+    for (User user: request.getUsersList()) {
+      mutations.add(
+          Mutation.newInsertBuilder("sometable")
+            .set("name")
+            .to(user.getName())
+            .set("age")
+            .to(Long.toString(user.getAge()))
+            .build());
+    }
+    dbClient.write(mutations);
+
+    EmptyResponse reply = EmptyResponse.newBuilder().build();
+    responseObserver.onNext(reply);
+    responseObserver.onCompleted();
+  }
+
+  public void update(UpdateQuery request, StreamObserver<EmptyResponse> responseObserver) {
+    System.out.println("update has been called");
+
+    dbClient
+      .readWriteTransaction()
+      .run(
+          new TransactionCallable<Void>() {
+            @Override
+            public Void run(TransactionContext transaction) throws Exception {
+              List<Statement> stmts = new ArrayList<Statement>();
```

Comment:
per google style use unabbreviated variable names; here statements instead of stmts

Reply:
also <Statement> --> <>